plugin/utils/casing_utils.py

```diff
@@ -32,6 +32,35 @@ def get_lower(s):
     return s[:1].lower() + s[1:]
 
 
+def to_snake(s):
```

Comment:
is this for translating a CamelCase Message name to a snake case name in the gapic config?

if so, i personally very much prefer to just keep the strings the same between the proto file resource definitions and the GAPIC config resource entity_names. This makes it easy for gapic-generator to cross-reference the GAPIC config resource entity_name with the proto resource `symbol` or parent message type.

otherwise, i have to implement the same `to_snake` method in gapic-generator...

Reply:
> if so, i personally very much prefer to just keep the strings the same between the proto file resource definitions and the GAPIC config resource entity_names

The goal is to keep the strings the same. I thought they _were_ in snake now, which is why I set them that way. I was looking [here](https://github.com/googleapis/gapic-generator/blob/master/src/test/java/com/google/api/codegen/testsrc/common/library_gapic.yaml#L72-L79), and these looked like snake to me.

Am I wrong? Totally willing to change it if so (and I should catch it when I start comparing files).

Reply:
ok yeah rethinking this, i think i was wrong. we aren't spitting out a real GAPIC config v2 in this plugin for use in gapic-generator, so going along with the snake_case convention in GAPIC config v1 is safe